jobs/cache_maintenance/tests/test_discussions.py

```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import pytest
+
+from cache_maintenance.discussions import (
+    create_link,
+    limit_to_one_dataset_per_namespace,
+)
+
+
+@pytest.mark.parametrize(
+    "datasets, valid_expected_datasets",
+    [
+        (set(), [set()]),
+        ({"a/b"}, [{"a/b"}]),
+        ({"a"}, [set()]),
+        ({"a/b/c"}, [set()]),
+        ({"a/b", "a/b"}, [{"a/b"}]),
+        ({"a/b", "a/c"}, [{"a/b"}, {"a/c"}]),
+        ({"a/b", "b/b"}, [{"a/b", "b/b"}]),
+        ({"a/b", "b"}, [{"a/b"}]),
+    ],
+)
+def test_limit_to_one_dataset_per_namespace(datasets: set[str], valid_expected_datasets: list[set[str]]) -> None:
```

Comment:
nit: Can you add test for post_messages function (maybe in another PR)?

Reply:
it was not a "nit" at all... I found and fixed innumerable bugs.